skstan/base.py

```diff
@@ -1,13 +1,12 @@
 from abc import ABCMeta, abstractstaticmethod
-from collections import UserDict
+from typing import Dict
 
 
-class BaseStanData(UserDict):
+class BaseStanData(Dict):
     def append(self, **kwargs):
-        self.data.update(kwargs)
-        return BaseStanData(
-            self.data
-        )
+        data = self.copy()
+        data.update(kwargs)
+        return data
```

Comment:
If you want to create completely immutable dict, `__setitem__` and `__delitem__` should be implemented and raise Exception.